src/node.cc

```diff
@@ -1216,6 +1238,24 @@ static ExitCode StartInternal(int argc, char** argv) {
 }
 
 int Start(int argc, char** argv) {
+  // Repeats argv[0] at position 1 on argv as a replacement for the missing
+  // entry point file path.
+  if (FindSingleExecutableCode() != nullptr) {
```

Comment:
Is this needed? I wouldn't expect `--compile` necessarily be the first argument. Do we need to remove it from the chain of argv? (How we do this for other internal arguments, such as `--trace_gc`? Afaik it doesn't get removed from the execution (argv) chain.

Reply:
Yes, this is needed to check if the binary is an SEA or just plain Node.js. It's needed regardless of having `--compile` in Node.js to change the way in which Node.js would execute if it finds an injected resource.